### ISSUE TITLE
Compiler allows to pass readonly object to function as non-readonly parameter

### ISSUE THREAD

**Issue by ilyub** [microsoft/TypeScript#45243]
# Bug Report

### 🔎 Search Terms

Compiler allows to pass readonly object to function as non-readonly parameter

### 🕗 Version & Regression Information

4.3.5

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgLhgJwKZwCbgDYE8ZgCuAtgEarIDaAujALww0DcAUC6JLKUgN4rpYweGADc42QqiREyFGAF96MPmIlSYARgWsWAM0JhgUAJbgYugBQAPaSXJVqAGhi5eo8ZNuzkCgJRIREGMMZXk2SzhnUl8mIA

### 💻 Code

```ts
const a: readonly number[] = [];

const b: { readonly value: number } = { value: 1 };

function f(x: number[], y: { value: number }): void {}

f(a, b);
```

### 🙁 Actual behavior

Shows error for the first parameter, but not for the second.

### 🙂 Expected behavior

Should show error for both parameters.

**Comment by MartinJohns** [comment]
Another duplicate of #13347.
